src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,384 @@
+			std::vector<Utility::NormalizedString> notFoundVersion;
+			FindMissing(versionsResultSet, minVersions, notFoundVersion);
+
+			std::for_each(
+				notFoundVersion.begin(),
+				notFoundVersion.end(),
+				[&](Utility::NormalizedString& version) { versionsResultSet[version] = VersionTable::EnsureExists(connection, version); });
+
+			StatementBuilder insertBuilder;
+			insertBuilder.InsertInto(s_DependenciesTable_Table_Name)
+				.Columns({ s_DependenciesTable_Manifest_Column_Name, s_DependenciesTable_MinVersion_Column_Name, s_DependenciesTable_PackageId_Column_Name })
+				.Values(Unbound, Unbound, Unbound);
+			SQLite::Statement insert = insertBuilder.Prepare(connection);
+
+			// TODO: use insert with multiple values, rather than run the query in a loop.
+			for (const auto& dep : dependencies)
+			{
+				insert.Reset();
+				insert.Bind(1, manifestRowId);
+				insert.Bind(2, versionsResultSet[dep.MinVersion.value().ToString()]);
+				insert.Bind(3, idResultSet[dep.Id]);
+
+				insert.Execute(true);
+			}
+		}
+	}
+
+	std::string_view DependenciesTable::TableName()
+	{
+		return s_DependenciesTable_Table_Name;
+	}
+
+	std::map<std::string, SQLite::rowid_t> SelectIdsByValues(const SQLite::Connection& connection, std::string_view tableName, std::string_view columnName, const std::vector<Utility::NormalizedString>& values)
+	{
+		using namespace Schema::V1_0;
+		using namespace SQLite::Builder;
+		const std::string rowIdColumnName = "rowid";
+
+		std::map<std::string, SQLite::rowid_t> result;
+
+		StatementBuilder selectStatement;
+		selectStatement.Select({ rowIdColumnName, columnName }).From(tableName).Where(columnName).In(values.size());
+
+		SQLite::Statement select = selectStatement.Prepare(connection);
+		int bindIndex = 1;
+		for (const std::string& value : values)
+		{
+			select.Bind(bindIndex, value);
+			bindIndex++;
+		}
+
+		while (select.Step())
+		{
+			result.emplace(select.GetColumn<std::string>(1), select.GetColumn<SQLite::rowid_t>(0));
+		}
+
+		return result;
+	}
+
+	void DependenciesTable::Create(SQLite::Connection& connection)
+	{
+		using namespace SQLite::Builder;
+		SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createDependencyTable_v1_3");
+
+		StatementBuilder createTableBuilder;
+		createTableBuilder.CreateTable(TableName()).BeginColumns();
+		createTableBuilder.Column(IntegerPrimaryKey());
+
+		std::vector<DependenciesTableColumnInfo> dependenciesColumns =
+		{ 
+			{ s_DependenciesTable_Manifest_Column_Name },
+			{ s_DependenciesTable_MinVersion_Column_Name },
+			{ s_DependenciesTable_PackageId_Column_Name }
+		};
+
+		for (const DependenciesTableColumnInfo& value : dependenciesColumns)
+		{
+			createTableBuilder.Column(ColumnBuilder(value.Name, Type::Int64).NotNull());
+		}
+
+		createTableBuilder.EndColumns();
+
+		createTableBuilder.Execute(connection);
+
+		StatementBuilder createPKIndexBuilder;
```

Comment:
The actual primary key index would be on manifest id and package id both and would be unique.
The index we need for add and potentially for client use is on manifest id.
The index needed for remove is on package id.

I would probably add all three so that it is easier to decide not to drop them in the future (although would still require a minor version update).